src/core/src/package_managers/ZypperPackageManager.py

```diff
@@ -79,6 +85,67 @@ def invoke_package_manager(self, command):
             self.set_package_manager_setting(Constants.PACKAGE_MGR_SETTING_REPEAT_PATCH_OPERATION, True)
         return out
 
+    """
+        When given a string containing a pid, returns a multi-line string with a list of parent/child processes for that pid.
+        See pattern matching inside the function for details on specific strings that are accepted.
+        
+        Example:
+            input: 
+                message (string): Output from package manager: | System management is locked by the application with pid 7914 (/usr/bin/zypper).
+                
+            returns (string):
+                  PID CMD
+                 7736 /bin/bash
+                 7912  \_ python3 package_test.py
+                 7913  |   \_ sudo LANG=en_US.UTF8 zypper --non-interactive update --dry-run bind-utils
+                 7914  |       \_ zypper --non-interactive update --dry-run bind-utils
+                 7982  |           \_ /usr/bin/python3 /usr/lib/zypp/plugins/urlresolver/susecloud
+                 7984  |               \_ /usr/bin/python3 /usr/bin/azuremetadata --api latest --subscriptionId --billingTag --attestedData --signature
+                 7986  \_ python3 package_test.py
+                 8298      \_ sudo LANG=en_US.UTF8 zypper --non-interactive update --dry-run grub2-i386-pc
+    """
```

Comment:
Ideally we shouldn't need the entire process tree info. Just details about the specific process mentioned in output should be good. For eg in this case just the process details for pid 7914. 

For now, let's keep this as is but add a todo comment in this function, stating we may want to revisit this later are reduce the result to specific process details only. 

Move the entire comment within the function

Reply:
Agreed. There might be a better way that I didn't see to get detailed information about a specific process, but I found that the process tree was the best for now because the output would print `/usr/bin/zypper` for the process name no matter if zypper was executed from the python script or from the terminal.

Moved the comment and added a todo.